File: src/main/java/com/github/frcsty/frozenactions/actions/broadcast/BroadcastAction.java
package com.github.frcsty.frozenactions.actions.broadcast;

import com.github.frcsty.frozenactions.actions.Action;
import com.github.frcsty.frozenactions.actions.ActionContext;
import com.github.frcsty.frozenactions.util.Extensions;
import org.bukkit.Bukkit;
import org.jetbrains.annotations.NotNull;

public final class BroadcastAction implements Action {

    @Override
    public @NotNull String getId() {
        return "BROADCAST";
    }

    @Override
    public void run(final @NotNull ActionContext context) {
        Bukkit.getOnlinePlayers().forEach(player ->
                Extensions.sendTranslatedMessage(player, context.getData())
        );
    }
}

File: src/main/kotlin/com/github/frcsty/frozenactions/actions/broadcast/BroadcastAction.kt
package com.github.frcsty.frozenactions.actions.broadcast

import com.github.frcsty.frozenactions.actions.Action
import com.github.frcsty.frozenactions.util.sendTranslatedMessage
import org.bukkit.Bukkit
import org.bukkit.entity.Player

object BroadcastAction : Action {
    override val id = "BROADCAST"

    override fun run(player: Player, data: String) = Bukkit.getServer().onlinePlayers.forEach { it.sendTranslatedMessage(data) }
}